DataGrid: add Column Chooser and Header Filter demos (#2572)
DevExpress/devextreme-demos: JSDemos/Demos/DataGrid/UsingHeaderFilter/description.md

> A header filter allows users to filter rows by values selected in a pop-up menu. To display this menu, users can click header filter icons in column headers.

Reviewer: In the second sentence you start in singular "this menu", but then go on to plural "filter icons in column headers". It's best to be consistent. 

```suggestion
A header filter allows users to filter rows by values selected in a pop-up menu. To display such a menu, users can click header a filter icon in a column header.
```

> A header filter allows users to filter rows by values selected in a pop-up menu. To display such a menu, users can click header a filter icon in a column header.

Reviewer 2: ```suggestion
A header filter allows users to filter rows by values selected in a pop-up menu. To display such a menu, users can click a header filter icon in a column header.
```

### Display Header Filter Icons

Assign **true** to the **headerFilter**.[visible](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/headerFilter/#visible) property to display header filter icons for all columns. To hide the icon for a specific column, set the column's [allowHeaderFiltering](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#allowHeaderFiltering) property to **false**.

> ### Enable Search Within Header Filters

Reviewer: ```suggestion
### Enable Search UI Within Header Filters
```

Users can search values within a header filter. To configure the search panel, define the [search](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/headerFilter/search/) property in the global [headerFilter](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/headerFilter/) object or in a **columns**.[headerFilter](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/headerFilter/) object. 

> You can compare the search string against specific fields in your data objects. For this purpose, apply the **search**.[searchExpr](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/headerFilter/search/#searchExpr) property. In this example, the search string is compared against the "City" and "State" fields.

Reviewer: I don't understand how this works. If I'm searching in a header filter, then that filter is already populated with values from some field. Why would I want to specify a different field for search? 

Let's say I clicked the filter icon in the "Country" column. The values in the header filter are Brazil, Spain, Portugal. I also enabled the search panel. Now I set `search.searchExpr` to compare my search string against the City field. Not sure how all of this makes sense. Now I can type in Rio or Madrid and find something?

Author: Ok, let's try to figure out. For example, you have the following data: **[{ city: 'London', country: 'United Kingdom' }, { city: 'Berlin', country: 'German' }]**. If you specify the columns[].headerFilter.**searchExpr: ['City', 'Country']** expression, in the header filter's search bar you can enter 'London' or 'United Kingdom' to get the 'London' result. So, if you enter just country name, you get all cities of this country.

Author: @Abadzhev Please, take a look at my answer)

Reviewer: What's the default behavior? Does the search include all fields, regardless of the column where I invoked the header filter?

Does the search panel even belong to a certain header filter? Maybe I just need to see a screenshot or demo.  

Here's what bothers me. Let's say I'm an end user. If I want to display all cities in a specific country. I would just search for a country. I would do that in the "Country" column. Entering a country name into the City column seems very weird to me. Why would anyone set up the control like that? :-)

The only way how all of this might make sense, is if the grid displayed just one single search panel... But you do mention that a search panel belongs to a header filter (a specific column). 

If you have a link to a demo I can see - it would be great.

Author: You can check the demo here: http://az-jsserver.corp.devexpress.com:81/Demos-23.1/WidgetsGallery/Demo/DataGrid/UsingHeaderFilter/

Reviewer: How about this version? 

```suggestion
The search panel's default behavior is to look for values only within the same data field. You can include additional fields into the search. Specify the **search**.[searchExpr](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/headerFilter/search/#searchExpr) property. For example, this demo allows you to enter a state name into the City column's header filter. You can then see a list of all cities within the specified state, and then select city names that you want to use as a filter. 
```

Reviewer: How about this version? 

```suggestion
The search panel's default behavior is to look for values only within the same data field. You can include additional fields into the search. Specify the **search**.[searchExpr](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/headerFilter/search/#searchExpr) property. For example, this demo allows you to enter a state name into the City column's header filter. You can then see a list of all cities within the specified state, and then select city names that you want to use as a filter. 
```

Reviewer: How about this version? 

```suggestion
The search panel's default behavior is to look for values only within the same data field. You can include additional fields into the search. Specify the **search**.[searchExpr](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/headerFilter/search/#searchExpr) property. For example, this demo allows you to enter a state name into the City column's header filter. You can then see a list of all cities within the specified state, and select city names that you want to use as a filter. 
```

Users can search values within a header filter. To configure the search panel, define the [search](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/headerFilter/search/) property in the global [headerFilter](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/headerFilter/) object or in a **columns**.[headerFilter](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/headerFilter/) object. 

> The search panel's default behavior is to look for values only within the same data field. You can include additional fields into the search. Specify the **search**.[searchExpr](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/headerFilter/search/#searchExpr) property. For example, this demo allows you to enter a state name into the City column's header filter. You can then see a list of all cities within the specified state, and select city names that you want to use as a filter. 

Reviewer 3: The search panel's default behavior looks for values only within the same data field. You can include additional fields in the search. Specify the **search**.[searchExpr] property. For example, this demo allows you to enter a state name in the City column's header filter. You can then see a list of all cities within the specified state, and select city names that you want to use as a filter.
